<section class="section projects-section" id="projects">
    <h2 class="section-title"><span class="icon-holder"><i class="fas fa-archive"></i></span>Projects</h2>
    <div class="intro">
        <p>This is an (incomplete) list of projects I worked on or that I am still working on.</p>
    </div><!--//intro-->

    <div class="accordion" id="ProjectsAccordion">
        <!--  Item      -->
        <div class="accordion-item" id="OpenSource">
            <h6 class="accordion-header" id="ProjectOpenSource">
                <button aria-controls="collapseProjectOpenSource" aria-expanded="true"
                        class="accordion-button job-title"
                        data-bs-target="#collapseProjectOpenSource" data-bs-toggle="collapse" type="button">
                    Open Source Contributions (ongoing)
                </button>
            </h6>
            <div aria-labelledby="ProjectOpenSource" class="accordion-collapse collapse show"
                 data-bs-parent="#ProjectsAccordion"
                 id="collapseProjectOpenSource">
                <div class="accordion-body">
                    <p>I contribute and update conda-forge recipes.</p>
                    <span class="badge bg-light text-dark">conda-forge</span>
                    <hr>
                    <p>In my day-to-day work I rely on a lot of open-source software, and I think it's important to
                        contribute to open-source projects myself.
                        So far my contributions mainly concern conda-forge recipes. Namely, I contributed the following:
                    <ul>
                        <li>addition of the PyPythia feedstock (maintainer)</li>
                        <li>addition of the apricot-select feedstock (maintainer)</li>
                        <li>update the pomegranate feedstock (maintainer)</li>
                        <li>update the scikit-allel feedstock (maintainer)</li>
                        <li>update the r-curl feedstock</li>
                    </ul>
                    Additionally, all my software projects are available open-source on GitHub.
                    </p>
                </div>
            </div>
        </div>
        <!-- //Item      -->
        <!--  Item      -->
        <div class="accordion-item" id="PandoraProject">
            <h6 class="accordion-header" id="ProjectsPandora">
                <button aria-controls="collapseProjectsPandora" aria-expanded="true"
                        class="accordion-button collapsed job-title"
                        data-bs-target="#collapseProjectsPandora" data-bs-toggle="collapse" type="button">
                    Pandora – Estimating the Uncertainty of Dimensionality Reduction
                </button>
            </h6>
            <div aria-labelledby="ProjectsPandora" class="accordion-collapse collapse"
                 data-bs-parent="#ProjectsAccordion"
                 id="collapseProjectsPandora">
                <div class="accordion-body">
                    <p>I implemented a Python tool to estimate the uncertainty of dimensionality reduction on population
                        genetics
                        datasets.</p>
                    <p>The tool is available on <a href=https://github.com/tschuelia/Pandora target="_blank">GitHub</a>
                        (WIP).</p>
                    <p>The respective preprint publication will be available soon.</p>
                    <span class="badge bg-light text-dark">Python</span>
                    <span class="badge bg-light text-dark">Eigensoft</span>
                    <span class="badge bg-light text-dark">Plotly</span>
                    <span class="badge bg-light text-dark">Pandas</span>
                    <span class="badge bg-light text-dark">Scikit-Learn</span>
                    <span class="badge bg-light text-dark">GitHub Actions</span>
                    <hr>
                    <p>
                        Dimensionality reduction techniques, for example Principal Components Analysis (PCA) or
                        Multidimensional scaling (MDS) are frequently used in population genetics and ancient DNA
                        studies.
                        To the best of my knowledge, there exists no standard in either field to report an estimate
                        of the uncertainty or instability of such methods. Without such an estimate, the accuracy
                        of a conclusion based on dimensionality reduction cannot be assessed by the reader of the study.
                        In my current work, I'm implementing the Python
                        framework Pandora that is able to provide such an uncertainty for a given genotype dataset
                        based on bootstrapping.
                        Pandora supports stability analyses for PCA and MDS analyses.
                        In addition to an overall stability of the dataset, Pandora can also compute the stability of
                        subsequent K-Means clustering analysis, as well as bootstrap support values for all samples in
                        the input dataset.
                    </p>
                    <p>The tool is unit-tested and I set up CI using GitHub actions.</p>
                </div>
            </div>
        </div>
        <!-- //Item      -->

        <!--  Item      -->
        <div class="accordion-item" id="SimulationsProject">
            <h6 class="accordion-header" id="ProjectsSimulation">
                <button aria-controls="collapseProjectsSimulation" aria-expanded="false"
                        class="accordion-button collapsed job-title"
                        data-bs-target="#collapseProjectsSimulation" data-bs-toggle="collapse" type="button">
                    Simulations of Sequence Evolution: how (un)realistic they really are and why.
                </button>
            </h6>
            <div aria-labelledby="ProjectsSimulation" class="accordion-collapse collapse"
                 data-bs-parent="#ProjectsAccordion"
                 id="collapseProjectsSimulation">
                <div class="accordion-body">
                    <p>We provide empirical evidence that current state-of-the-art models of sequence evolution cannot
                        reproduce
                        empirical-like evolutionary data.</p>
                    <p>The publication is available as <a
                            href="https://doi.org/10.1101/2023.07.11.548509"
                            target="_blank">preprint in BioRxiv</a>.</p>
                    <span class="badge bg-light text-dark">Python</span>
                    <span class="badge bg-light text-dark">Pandas</span>
                    <span class="badge bg-light text-dark">Scikit-Learn</span>
                    <span class="badge bg-light text-dark">LightGBM</span>
                    <hr>
                    <p>
                        Simulating sequence evolution plays an important role in the development and evaluation of
                        phylogenetic inference tools. Naturally, the simulated data needs to be as realistic as possible
                        to be indicative of the performance of the developed tools on empirical data. Over the years,
                        numerous phylogenetic sequence simulators, employing various models of evolution, have been
                        published with the goal to simulate such empirical-like data. In this study, we simulated DNA
                        and protein Multiple Sequence Alignments (MSAs) under increasingly complex models of evolution
                        with and without insertion/deletion (indel) events using a state-of-the-art sequence simulator.
                        We assessed their realism by quantifying how well supervised learning methods are able to
                        predict whether a given MSA is simulated or empirical.
                    </p>
                    <p>
                        Our results show that we can distinguish between empirical and simulated MSAs with high accuracy
                        using two distinct and independently developed classification approaches across all tested
                        models of sequence evolution. Our findings suggest that the current state-of-the-art models fail
                        to accurately replicate the process of evolution.
                    </p>
                    <p>This work is a joint work with my colleagues at HITS and a team of researchers in Lyon. Johanna
                        Trost, Dimitri Höhler, and I contributed all equally to this work.
                    </p>
                </div>
            </div>
        </div>
        <!-- //Item      -->

        <!--  Item      -->
        <div class="accordion-item" id="PythiaProject">
            <h6 class="accordion-header" id="ProjectsOne">
                <button aria-controls="collapseProjectsOne" aria-expanded="false"
                        class="accordion-button collapsed job-title"
                        data-bs-target="#collapseProjectsOne" data-bs-toggle="collapse" type="button">
                    Pythia – Predicting the Difficulty of Phylogenetic Analyses.
                </button>
            </h6>
            <div aria-labelledby="ProjectsOne" class="accordion-collapse collapse"
                 data-bs-parent="#ProjectsAccordion"
                 id="collapseProjectsOne">
                <div class="accordion-body">
                    <p>
                        I implemented a Python library for predicting the difficulty of multiple sequence alignments in
                        phylogenetics.
                    </p>
                    <p>PyPythia is available on <a href="https://github.com/tschuelia/PyPythia">GitHub</a>.</p>
                    <p>The peer-reviewed publication is available <a
                            href="https://doi.org/10.1093/molbev/msac254"
                            target="_blank">in Molecular Biology and Evolution</a>.</p>
                    <span class="badge bg-light text-dark">Python</span>
                    <span class="badge bg-light text-dark">Pandas</span>
                    <span class="badge bg-light text-dark">Scikit-Learn</span>
                    <span class="badge bg-light text-dark">LightGBM</span>
                    <span class="badge bg-light text-dark">TreeLite</span>
                    <span class="badge bg-light text-dark">Snakemake</span>
                    <span class="badge bg-light text-dark">RAxML-NG</span>
                    <span class="badge bg-light text-dark">GitHub Actions</span>
                    <hr>
                    <p>
                        Phylogenetic analyses under the Maximum Likelihood model are time and resource intensive.
                        To adequately capture the vastness of tree space, one needs to infer multiple independent trees.
                        On some datasets, multiple tree inferences converge to similar tree topologies, on others to
                        multiple,
                        topologically highly distinct yet statistically indistinguishable topologies.
                        At present, no method exists to quantify and predict this behavior.
                        We introduce a method to quantify the degree of difficulty for analyzing a dataset and present
                        Pythia, a Random Forest Regressor that accurately predicts this difficulty. Pythia predicts the
                        degree of
                        difficulty of analyzing a dataset prior to initiating Maximum Likelihood based tree inferences.
                        The prediction and computation of all required attributes is substantially faster than a single
                        Maximum Likelihood tree inference.
                        Pythia can be used to increase user awareness with respect to the amount of signal and
                        uncertainty to be
                        expected in phylogenetic analyses, and hence inform an appropriate (post-)analysis setup.
                        Further, it can be used to select appropriate search algorithms for easy-, intermediate-,
                        and hard-to-analyze datasets.
                    </p>
                    <p>In the current version of Pythia we replaced the Random Forest Regressor with boosted trees
                        implemented in
                        LightGBM.</p>
                    <p>PyPythia is unit-tested and I set up CI using GitHub Actions.</p>
                    <p>Pythia is available as open source software libraries in C (CPythia) and Python (PyPythia).</p>
                    <p>
                        CPythia is also available on <a href="https://github.com/tschuelia/CPythia">GitHub</a>.
                        CPythia can only be used in conjunction with <a
                            href="https://codeberg.org/Exelixis-Lab/coraxlib">Coraxlib</a>.
                    </p>

                </div>
            </div>
        </div>
        <!-- //Item      -->

        <!--  Item      -->
        <div class="accordion-item" id="NumericalAnalysis">
            <div class="accordion-header" id="ProjectsSeven">
                <button aria-controls="collapseProjectsSeven" aria-expanded="false"
                        class="accordion-button collapsed job-title"
                        data-bs-target="#collapseProjectsSeven" data-bs-toggle="collapse" type="button">
                    Empirical Numerical Properties of Maximum Likelihood Phylogenetic
                    Inference
                </button>
            </div>
            <div aria-labelledby="ProjectsSeven" class="accordion-collapse collapse"
                 data-bs-parent="#ProjectsAccordion"
                 id="collapseProjectsSeven">
                <div class="accordion-body">
                    <p>I performed large scale numerical analysis of the influence of threshold parameters on the
                        runtime and
                        results of Maximum Likelihood phylogenetic inferences.</p>
                    <p>The peer-reviewed publication is available <a
                            href="https://doi.org/10.1093/bioadv/vbad124"
                            target="_blank">in Bioinformatics Advances</a>.</p>
                    <span class="badge bg-light text-dark">Python</span>
                    <span class="badge bg-light text-dark">Snakemake</span>
                    <span class="badge bg-light text-dark">Plotly</span>
                    <span class="badge bg-light text-dark">RAxML-NG</span>
                    <span class="badge bg-light text-dark">IQ-Tree</span>
                    <span class="badge bg-light text-dark">FastTree</span>
                    <hr>
                    <p>
                        Maximum Likelihood (ML) is a widely used phylogenetic inference model. ML implementations
                        heavily rely on numerical optimization routines that use internal numerical thresholds to
                        determine convergence. We systematically analyze the impact of these threshold settings on the
                        log-likelihood and runtimes for ML tree inferences with RAxML-NG, IQ-TREE, and FastTree on
                        empirical datasets. We provide empirical evidence that we can substantially accelerate tree
                        inferences with RAxML-NG and IQ-TREE by changing the default values of two such numerical
                        thresholds. At the same time, altering these settings does not significantly impact the quality
                        of the inferred trees. We further show that increasing both thresholds accelerates the RAxML-NG
                        bootstrap without influencing the resulting support values. For RAxML-NG, increasing the
                        likelihood thresholds ϵ<sub>LnL</sub> and ϵ<sub>brlen</sub> to 10 and 10<sup>3</sup>
                        respectively results
                        in an average tree
                        inference speedup of 1.9 ± 0.6 on Data collection 1, 1.8 ± 1.1 on Data collection 2, and 1.9 ±
                        0.8 on Data collection 2 for the RAxML-NG bootstrap. Increasing the likelihood threshold ϵ<sub>LnL</sub>
                        to
                        10 in IQ-TREE results in an average tree inference speedup of 1.3 ± 0.4 on Data collection 1 and
                        1.3 ± 0.9 on Data collection 2.
                    </p>
                    <p>Our research comprises four studies, I conducted Study 1 during my Master's thesis with the CME
                        group at HITS.
                        You can find the full thesis <a
                                href="https://cme.h-its.org/exelixis/pubs/masterJulia.pdf">here</a></p>.
                </div>
            </div>
        </div>
        <!-- //Item      -->

        <!--  Item      -->
        <div class="accordion-item">
            <div class="accordion-header" id="ProjectsTwo">
                <button aria-controls="collapseProjectsTwo" aria-expanded="false"
                        class="accordion-button collapsed job-title"
                        data-bs-target="#collapseProjectsTwo" data-bs-toggle="collapse" type="button">
                    Diary Website
                </button>
            </div>
            <div aria-labelledby="ProjectsTwo" class="accordion-collapse collapse"
                 data-bs-parent="#ProjectsAccordion"
                 id="collapseProjectsTwo">
                <div class="accordion-body">
                    <p>I implemented a private web-based diary to keep track of vacations.</p>
                    <p>The source code is available <a href="https://github.com/tschuelia/diary"
                                                       target="_blank">on GitHub</a>.</p>
                    <span class="badge bg-light text-dark">Python</span>
                    <span class="badge bg-light text-dark">Django</span>
                    <span class="badge bg-light text-dark">HTML</span>
                    <span class="badge bg-light text-dark">CSS</span>
                    <span class="badge bg-light text-dark">Bootstrap</span>
                    <hr>
                    <p>
                        To remember vacations, holidays and trips I implemented a diary website using the django web
                        framework.
                        This website also stores images associated with certain trips and image descriptions, so I can
                        show the best images to friends and family. It also includes a map, so I can see where
                        on this beautiful earth I have already travelled to ☺︎
                        In its latest version, I can also upload files, e.g. GPS tracks of hikes.
                    </p>
                </div>
            </div>
        </div>
        <!-- //Item      -->

        <!--  Item      -->
        <div class="accordion-item">
            <div class="accordion-header" id="ProjectsThree">
                <button aria-controls="collapseProjectsThree" aria-expanded="false"
                        class="accordion-button collapsed job-title"
                        data-bs-target="#collapseProjectsThree" data-bs-toggle="collapse" type="button">
                    Master Courses Organization Tool
                </button>
            </div>
            <div aria-labelledby="ProjectsThree" class="accordion-collapse collapse"
                 data-bs-parent="#ProjectsAccordion"
                 id="collapseProjectsThree">
                <div class="accordion-body">
                    <p>I implemented a web-based tool to organize courses for the computer science master's degree at
                        KIT.</p>
                    <p>The source code is available <a href="https://github.com/tschuelia/master-organization"
                                                       target="_blank">on GitHub</a>.</p>
                    <span class="badge bg-light text-dark">Python</span>
                    <span class="badge bg-light text-dark">Django</span>
                    <span class="badge bg-light text-dark">HTML</span>
                    <span class="badge bg-light text-dark">CSS</span>
                    <span class="badge bg-light text-dark">Bootstrap</span>
                    <hr>
                    <p>
                        For my master’s studies at KIT I implemented a web-based tool to organize the courses I
                        intend to take. The tool checks if the planned courses satisfy all requirements to get the
                        master’s degree. It further shows a schedule of exam dates, as well as an overview of all
                        grades.
                        Since the computation of the final grade at KIT is a bit tedious, I also implemented an
                        automatic
                        computation of the final grade, as well as the average grades per module.
                    </p>
                </div>
            </div>
        </div>
        <!-- //Item      -->

        <!--  Item      -->
        <div class="accordion-item">
            <div class="accordion-header" id="ProjectsFour">
                <button aria-controls="collapseProjectsFour" aria-expanded="false"
                        class="accordion-button collapsed job-title"
                        data-bs-target="#collapseProjectsFour" data-bs-toggle="collapse" type="button">
                    Markerless Patient Tracking (Bachelor’s Thesis)
                </button>
            </div>
            <div aria-labelledby="ProjectsFour" class="accordion-collapse collapse"
                 data-bs-parent="#ProjectsAccordion"
                 id="collapseProjectsFour">
                <div class="accordion-body">
                    <p>I implemented a markerless patient movement tracking algorithm for cochlear implant
                        surgeries.</p>
                    <p>The source code is available <a href="https://github.com/tschuelia/Image-Based-Tracking"
                                                       target="_blank">on GitHub</a>.</p>
                    <p>The peer-reviewed paper is available <a href="https://doi.org/10.3389/fsurg.2021.742160">Frontiers
                        in Surgery</a>.</p>
                    <span class="badge bg-light text-dark">Python</span>
                    <span class="badge bg-light text-dark">OpenCV</span>
                    <span class="badge bg-light text-dark">scikit-image</span>
                    <hr>
                    <p>
                        As my bachelor's thesis, at the Intelligent Process Automation and Robotics Lab (IPR) at KIT, I
                        implemented a system
                        for markerless patient movement tracking during cochlear implant surgeries. The approach uses
                        only
                        the images obtained by the microscopic camera used by the surgeon.
                        I implemented the algorithm using Python and the image processing frameworks OpenCV and
                        scikit-image. It also includes a neural network for semantic image segmentation.

                        My bachelor's thesis is available upon request only as it contains sensitive patient images.
                    </p>
                </div>
            </div>
        </div>
        <!-- //Item      -->

        <!--  Item      -->
        <div class="accordion-item">
            <div class="accordion-header" id="ProjectsFive">
                <button aria-controls="collapseProjectsFive" aria-expanded="false"
                        class="accordion-button collapsed job-title"
                        data-bs-target="#collapseProjectsFive" data-bs-toggle="collapse" type="button">
                    Recipe Website
                </button>
            </div>
            <div aria-labelledby="ProjectsFive" class="accordion-collapse collapse"
                 data-bs-parent="#ProjectsAccordion"
                 id="collapseProjectsFive">
                <div class="accordion-body">
                    <p>I implemented a website for storing and sharing recipes.</p>
                    <p>The source code is available <a href="https://github.com/tschuelia/cake_recipe_website"
                                                       target="_blank">on GitHub</a>.</p>
                    <span class="badge bg-light text-dark">Python</span>
                    <span class="badge bg-light text-dark">Django</span>
                    <span class="badge bg-light text-dark">HTML</span>
                    <span class="badge bg-light text-dark">CSS</span>
                    <span class="badge bg-light text-dark">Bootstrap</span>
                    <hr>
                    <p>This project was inspired by my love for baking. It is meant for storing all the ideas and
                        recipes
                        of me and my friends. Different users can create new recipes, categorize them or search for a
                        specific recipe.
                        The website also includes functionality to filter recipes by ingredients and categories.
                    </p>
                    <p>This website is an ongoing project and I keep adding new features. The latest feature I added
                        was a shopping list that is especially useful when planning to bake multiple different
                        recipes.</p>

                </div>
            </div>
        </div>
        <!-- //Item      -->

        <!--  Item      -->
        <div class="accordion-item">
            <div class="accordion-header" id="ProjectsSix">
                <button aria-controls="collapseProjectsSix" aria-expanded="false"
                        class="accordion-button collapsed job-title"
                        data-bs-target="#collapseProjectsSix" data-bs-toggle="collapse" type="button">
                    Lambda Calculus IDE (study project)
                </button>
            </div>
            <div aria-labelledby="ProjectsSix" class="accordion-collapse collapse"
                 data-bs-parent="#ProjectsAccordion"
                 id="collapseProjectsSix">
                <div class="accordion-body">
                    <p>We implemented a web-based Lambda Calculus IDE and interpreter for the programming paradigm
                        lecture at KIT.</p>
                    <p>The source code is available <a href="https://github.com/wavelength-ide/wavelength-ide"
                                                       target="_blank">here</a>.</p>
                    <span class="badge bg-light text-dark">Java</span>
                    <span class="badge bg-light text-dark">Google Web Toolkit</span>
                    <span class="badge bg-light text-dark">JavaScript</span>
                    <span class="badge bg-light text-dark">CSS</span>
                    <hr>
                    <p>As part of the undergraduate studies, we implemented a browser-based lambda calculus IDE and
                        interpreter in a team of six students. We used Java with the Google Web Toolkit, JavaScript and
                        CSS
                        and wrote about 10 000 lines of code plus 6000 lines of tests.
                        The project was graded with mark 1.0 (best possible grade).
                        I mostly implemented the interactive frontend to explore the lambda terms and their reduction
                        steps.
                        Also, I wrote lots of tests for the controller layer (MVC architecture).
                    </p>
                </div>
            </div>
        </div>
        <!-- //Item      -->
    </div>

</section><!--//section-->
